src/main/java/dev/the_fireplace/unforgivingvoid/mixin/ServerPlayerEntityMixin.java

```diff
@@ -36,22 +38,27 @@ private void tick(CallbackInfo callbackInfo) {
         DimensionConfig dimensionConfig = DIContainer.get().getInstance(DimensionConfigManager.class).getSettings(this.world.getRegistryKey().getValue());
         if (!this.world.isClient()
             && dimensionConfig.isEnabled()
-            && this.getBlockPos().getY() <= getBottomY(world) - dimensionConfig.getTriggerDistance()
+            && this.getBlockPos().getY() <= world.getBottomY() - dimensionConfig.getTriggerDistance()
             && !isInTeleportationState()
         ) {
             MinecraftServer server = getServer();
             if (server != null) {
                 UnforgivingVoidConstants.getLogger().debug(
                     "Player is below the minimum height. Teleporting to new dimension. Current position is {}, and current world is {}",
                     getBlockPos().toShortString(),
-                    getWorld().getRegistryKey().getValue()
+                    getServerWorld().getRegistryKey().getValue()
                 );
+
+                inTeleportationState = true;
```

Comment:
isInTeleportationState() is still shadowed. Does the new private variable correctly map to the shadowed variable outside the dev environment? If not, we might want to use a setter accessor to modify the existing one instead.

Reply:
This variable should be correctly mapped outside the dev env. I see no reason why it shouldn't be.

I kept `isInTeleportationState()`, as it's probably still best to check it through that.

Reply:
The reason I ask is this is a private variable with no annotation to indicate that it's shadowed. So I'm not sure if Mixin will consider that to be a shadowed variable and remap it accordingly, or treat it as a private helper variable like if it were named something that does not exist in the target class.

Reply:
It very much is shadowed. You must have misread the diff, as the `@Shadow` annotation was not removed.

Reply:
You're right, I misread it. My mistake.